app/controllers/admin/enterprises_controller.rb

```diff
@@ -171,8 +171,8 @@ def collection_actions
     end
 
     def load_methods_and_fees
-      @payment_methods = Spree::PaymentMethod.managed_by(spree_current_user).sort_by!{ |pm| [(@enterprise.payment_methods.include? pm) ? 0 : 1, pm.name] }
-      @shipping_methods = Spree::ShippingMethod.managed_by(spree_current_user).sort_by!{ |sm| [(@enterprise.shipping_methods.include? sm) ? 0 : 1, sm.name] }
+      @payment_methods = Spree::PaymentMethod.all.sort_by!{ |pm| [(@enterprise.payment_methods.include? pm) ? 0 : 1, pm.name] }
+      @shipping_methods = Spree::ShippingMethod.all.sort_by!{ |sm| [(@enterprise.shipping_methods.include? sm) ? 0 : 1, sm.name] }
```

Comment:
This will load all shipping methods available in the database blowing up the server. We should query the data instead, by means of an INNER JOIN.

Reply:
@sauloperez, for admin by default we are showing all the shipping methods, 
```
scope :managed_by, lambda { |user|
    if user.has_spree_role?('admin')
      scoped
    else
      joins(:distributors).
        where('distributors_shipping_methods.distributor_id IN (?)', user.enterprises).
        select('DISTINCT spree_shipping_methods.*')
    end
  }
```
for manager also we should show all the shipping methods, that was the issue. So I removed the managed_by scope.

I cant understand how can I fix the issue by using inner join, can you please guide on fixing the issue with more details.

Thank you.

Reply:
I see. What about the `else` branch's behavior? In the case of non-admin users, we want to show the shipping methods that the enterprise uses. This behavior will be gone if we do not call `:managed_by`. 

Regarding the INNER JOIN, nevermind. We should create another issue to solve it, which will probably mean changing the behavior a bit. Let's stick to the bug fix here.

Reply:
Yes @sauloperez, For non-admin users If we use managed_by scope it will only display the shipping/payment methods that are used by the enterprise, the method `load_methods_and_fees` in admin/enterprise-controller will be used by only admins & managers, for managers also we need to display all the shipping/payment methods instead of only the items used by the enterprise. As the managers are non-admin, for them this method will display only the shipping/payment methods used by the enterprise. Also due to this when a manager uncheck an shipping/payment methods they will disappear from the list.

So, instead of using the managed_by scope i used all.

Correct me if I am wrong and please let me know if there is any other way of fixing this.

Reply:
Yes @sauloperez, For non-admin users If we use managed_by scope it will only display the shipping/payment methods that are used by the enterprise, the method load_methods_and_fees in admin/enterprise-controller will be used by only admins & managers, for managers also we need to display all the shipping/payment methods instead of only the items used by the enterprise. As the managers are non-admin, for them this method will display only the shipping/payment methods used by the enterprise. Also due to this when a manager uncheck an shipping/payment methods they will disappear from the list.

So, instead of using the managed_by scope i used all.

Correct me if I am wrong and please let me know if there is any other way of fixing this.

Reply:
Thanks! Now I get it all. I don't think there's any other quick way to fix this for now.